src/core/qgis.h

```diff
@@ -233,6 +233,17 @@ inline double qgsRound( double x )
   return x < 0.0 ? std::ceil( x - 0.5 ) : std::floor( x + 0.5 );
 }
 
+/**
+ * Returns a double \a number, rounded (as close as possible) to the specified number of \a places.
+ *
+ * @note Added in QGIS 3.0
+ */
+inline double qgsRound( double number, double places )
+{
+  int scaleFactor = pow( 10, places );
```

Comment:
@m-kuhn, I think there is a flaw in qgsRound() that leads to incorrect rounding and/or overflow in some circumstances.
Could you please have a look at http://osgeo-org.1560.x6.nabble.com/QGIS-Developer-On-precision-in-qgsRound-3-3-0-master-td5377378.html and https://github.com/qgis/QGIS/pull/7836#issuecomment-419718554